src/engine/authentication.md

```diff
@@ -35,10 +35,24 @@ The HMAC algorithm implies that several CLs will be able to use the same key, an
 
 The `EL` and `CL` clients **MUST** accept a cli/config parameter: `jwt-secret`, which designates a file containing the hex-encoded 256 bit secret key to be used for verifying/generating JWT tokens.
 
-If such a parameter is not given, the client **SHOULD** generate such a token, valid for the duration of the execution, and store the hex-encoded secret as a `jwt.hex` file on the filesystem.  This file can then be used to provision the counterpart client.
-
 If such a parameter _is_ given, but the file cannot be read, or does not contain a hex-encoded key of `256` bits, the client should treat this as an error: either abort the startup, or show error and continue without exposing the authenticated port.
 
+If such a parameter is not given, the client **MUST** attempt to read the secret from the default paths defined below, in the order they are listed.  If a secret is found, but the file cannot be read, does not contain a hex-encoded key of `256` bit, or is rejected by the other client, the client should continue searching. The CL **MAY** search other locations, such as default EL data directories.
```

Comment:
As mentioned below, I'm not a fan of the **MUST** here.  A high security client may reasonably refuse to read secrets from disk and demand that they be read from a secure secret location in order to properly start.  Such a client shouldn't be in violation of the specification in that section.  I think a **SHOULD** would be appropriate here and address the need that I believe you are trying to address.

Reply:
I think a "high security" client should just flag this configuration with lots of warnings. If I knew all current clients would implement this if it is listed as **SHOULD**, I would be inclined to change it. But generally, I think this is an important UX improvement and we should force clients to follow suite.

Reply:
If it is sufficiently useful to users, then all clients will implement it without being forced to.  I generally am pretty 👎 on sacrificing security for usability.  We should always try to find ways to improve usability, but never at the cost of security.

A "high security client" should not even start if it is configured in a way that is insecure.  This sort of client is valuable for businesses and enterprise shops where they want to make sure human error cannot result in a security compromise.  Those clients wouldn't be using this feature at all, as they would have their own key management system that injects the keys just-in-time (likely as part of whatever process is launching the client).